hist/histv7/inc/ROOT/RHist.hxx

```diff
@@ -308,15 +308,29 @@ using RH3I = RHist<3, int, RHistStatContent>;
 using RH3LL = RHist<3, int64_t, RHistStatContent>;
 ///\}
 
+template <int DIMENSIONS, class PRECISION,
+          template <int D_, class P_> class... STAT_FIRST,
+          template <int D_, class P_> class... STAT_SECOND>
+void SameAxesOrThrow(RHist<DIMENSIONS, PRECISION, STAT_FIRST...> &first, const RHist<DIMENSIONS, PRECISION, STAT_SECOND...> &second)
+{
+   auto& firstImpl = *first.GetImpl();
+   const auto& secondImpl = *second.GetImpl();
+   for (int dim = 0; dim < DIMENSIONS; ++dim) {
+      if (!firstImpl.GetAxis(dim).HasSameBinningAs(secondImpl.GetAxis(dim))) {
+         throw std::runtime_error("Attempted to merge two RHists with incompatible axis binning!");
+      }
+   }
+}
+
 /// Add two histograms.
 ///
 /// This operation may currently only be performed if the two histograms have
 /// the same axis configuration, use the same precision, and if `from` records
 /// at least the same statistics as `to` (recording more stats is fine).
 ///
 /// Adding histograms with incompatible axis binning will be reported at runtime
-/// with an `std::runtime_error`. Insufficient statistics in the source
-/// histogram will be detected at compile-time and result in a compiler error.
+/// with an `std::runtime_error`. Incompatible statistics' size in the source
```

Comment:
It's not just about size, the actual constraint (as of the current code) is that "from" must record at least the same statistics as "to" or else the `toImpl.GetStat().Add(fromImpl.GetStat())` call will fail because the statistics which only exist in "to" will not find their counterpart in "from" and therefore the call to the corresponding virtual statistics class method in the implementation will fail.